url/src/quirks.rs

```diff
@@ -212,7 +212,10 @@ pub fn set_hostname(url: &mut Url, new_hostname: &str) -> Result<(), ()> {
         return Ok(());
     }
 
-    if let Ok((host, _remaining)) = Parser::parse_host(input, scheme_type) {
+    if let Ok((host, remaining)) = Parser::parse_host(input, scheme_type) {
+        if remaining.split_prefix(':').is_some() {
```

Comment:
```suggestion
        if remaining.starts_with(':') {
```

I think it's much clearer to use starts_with instead of split_prefix.